cmd/cnab-to-oci/fixup.go

```diff
@@ -59,7 +59,18 @@ func runFixup(opts fixupOptions) error {
 	if err != nil {
 		return err
 	}
-	if err := remotes.FixupBundle(context.Background(), &b, ref, resolver); err != nil {
+	eventsCh := make(chan remotes.FixupEvent)
+	eventsShown := make(chan struct{})
+	go func() {
+		defer close(eventsShown)
+		for ev := range eventsCh {
+			fmt.Fprintf(os.Stderr, "%s: source=%s, dest=%s, descriptor: %#v, message: %s, error: %s\n", ev.EventType, ev.SourceImage, ev.DestinationRef, ev.Descriptor, ev.Message, ev.Error)
+		}
+	}()
+	err = remotes.FixupBundle(context.Background(), &b, ref, resolver, eventsCh)
+	close(eventsCh)
```

Comment:
I would normally expect `FixupBundle` to do this when it has finished.

Reply:
Hmm, it is not FixupBundle that creates the channel. I think the best practice for closing channels, is to only do that in the same component that creates it in the first place. I might be wrong though.

Reply:
I guess it depends on whether you think of `FixupBundle` as taking ownership or not, I'm influenced by e.g. buildkit's `Solve` method into thinking it does, maybe that isn't the norm though?

I guess the actual norm is to return the chan from `FixupBundle`, I think that doesn't work here though since `FixupBundle` is a synchronous call.

Reply:
Yes, the synchronous call makes it non-obvious. I prefer the "whoever creates should close" approach (as actually, a chan can be kept open if it makes sense for the client code)